src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/BlazorDebugConfigurationProvider.ts

```diff
@@ -76,15 +79,16 @@ export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurati
             await this.vscodeType.debug.startDebugging(folder, app);
             if (process.platform !== 'win32') {
                 const terminate = this.vscodeType.debug.onDidTerminateDebugSession(async event => {
-                    const blazorDevServer = 'blazor-devserver.dll';
+                    const blazorDevServer = 'blazor-devserver\\.dll';
                     const dir = folder && folder.uri && folder.uri.fsPath;
-                    const launchedApp = configuration.hosted ? app.program : `${dir}.*${blazorDevServer}|${blazorDevServer}.*${dir}`;
+                    const regexEscapedDir = dir?.toLowerCase()?.replace(/\//g, '\\/');
```

Comment:
There's probably a node package to help with escaping strings for regex matching, but wanted to avoid adding an additional dependency for this limited use case. 

`toLowerCase` as file paths on MacOS are case **insensitive**, hence without it, regex matching was failing. I could've done a case insensitive regex match, however the general consensus online seemed to be `toLowerCase` was preferred.